Pinta.Gui.Widgets/Widgets/Canvas/CanvasWindow.cs

```diff
@@ -166,12 +209,41 @@ public void UpdateRulerRange (object? sender, EventArgs e)
 
 	private bool HandleScrollEvent (EventControllerScroll controller, EventControllerScroll.ScrollSignalArgs args)
 	{
-		// Allow the user to zoom in/out with Ctrl-Mousewheel
+		// Allow the user to zoom in/out with Ctrl-Mousewheel or Ctrl-two-finger-scroll
 		if (controller.GetCurrentEventState ().IsControlPressed ()) {
-			if (args.Dx > 0 || args.Dy < 0)
+			// "clicky" scroll wheels generate 1 or -1
+			if (args.Dx == 1 || args.Dy == -1) {
 				document.Workspace.ZoomInAroundCanvasPoint (current_canvas_pos);
-			else if (args.Dx < 0 || args.Dy > 0)
+				return true;
+			}
+			if (args.Dx == -1 || args.Dy == 1) {
 				document.Workspace.ZoomOutAroundCanvasPoint (current_canvas_pos);
+				return true;
+			}
+
+			// analog scroll wheels and scrolling on a touchpad generates a range of values constantly as the user scrolls
+			// this might feel "backwards" on a touchpad to some people
+			if (args.Dx > 0 || args.Dy < 0) {
+				if (cumulative_zoom_amount > 0)
+					cumulative_zoom_amount = 0;
+				cumulative_zoom_amount += args.Dx;
```

Comment:
With `Dx` and `Dy` having opposite signs, can this counteract the accumulated value somewhat?

Reply:
I'll take a look at this in the morning. You're probably right, although I have no clue if `Dx` ever changes from zero.